tutorials/04 - DistortionEffectDemo.md

```diff
@@ -0,0 +1,138 @@
+# 04 - Distortion Effect
+
+In this tutorial, we're going to analyze the structure of an audio effect application, by applying distortion to a looping audio file.
+
+## Main Tracktion Engine Classes
+#### Engine
+Every tracktion Engine app starts with an 'Engine' instance. This performs the initialisation and shutdown of the engine.
+
+#### Edit
+The 'Edit' is the container for a playable arrangement. it holds tracks, tempo sequences, Racks etc.
+
+#### Track
+'Edit' contains a list of 'Track's. This project makes use of the 'AudioTrack' which outputs audio.
+
+#### Clip
+'Clip' is the base class for clips that live in 'Track's. 'Clip's can be MIDI, audio, step or the edit-in-an-edit 'editClip.'
+
+
+## Main JUCE Classes
+#### Plugin
+
+## DistortionEffectDemo.h
+
+
+## DistortionPlugin Class
+
+### DistortionPlugin() Constructor
+
+- In the constructor, we initialise the undo manager to enable users to undo plugin value changes, and refer each plugin value to the undo manager.
```

Comment:
This isn't really correct. We're not initialising the UndoManager.
We're referring the gainValue CachedValue member to the plugin's `state` gain` property, using the Edit's undo manager and providing a default value of 1.